code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthTokenDaoImpl.java

```diff
@@ -0,0 +1,84 @@
+package io.cattle.platform.iaas.api.auth.dao.impl;
+
+import static io.cattle.platform.core.model.tables.AuthTokenTable.AUTH_TOKEN;
+
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.dao.GenericResourceDao;
+import io.cattle.platform.core.model.AuthToken;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.dao.AuthTokenDao;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.filter.apikey.ApiKeyFilter;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+public class AuthTokenDaoImpl extends AbstractJooqDao implements AuthTokenDao{
+
+    @Inject
+    GenericResourceDao resourceDao;
+    @Inject
+    ObjectManager objectManager;
+    @Inject
+    ObjectProcessManager objectProcessManager;
+
+    @Override
+    public AuthToken getTokenById(String key) {
```

Comment:
We should rename it to getTokenByKey() rather than getTokenById()
since both 'key' and 'id' exist in the DB table and we're actually looking it up by 'key'